docs/syntax-octal-literals.md

```diff
@@ -0,0 +1,33 @@
+# Octal integer literals
+
+## Summary
+
+This RFC proposes adding support for defining octal literals in the form `0oNNN...` - for example, `0o777` evaluates to 511 in decimal. 
+
+## Motivation
+
+Some legacy systems use octal extensively, and converting from octal to other bases to interface with them can be a chore. For example, the PDP-11 uses octal extensively - its instruction encodings are a natural fit for octal as they are arranged in groups of 3 bits, and octal is used to represent addresses & words in nearly all materials.
+
+Unix also uses octal to represent file permissions (e.g. in commands like `chmod 755`), and this is perhaps the most recognisable use of it today.
+
+Many other languages (Python, JavaScript, Rust...) also support octal literals in this style, so it allows for more straightforward porting of code.
+
+The audience for this feature is admittedly small, but the parser provisions for supporting hexadecimal and binary literals extend easily onto adding octal literals.
```

Comment:
I don't know if this is actually a motivation, it's just a description of why you'd want to use octals in general. It doesn't answer why Luau should have them beyond "it makes porting code easier".